### ISSUE TITLE
Advise against wrapAll in NAMD

### ISSUE THREAD

**Issue by jhenin** [colvars/colvars#90]
We've seen many cases of NAMD colvar simulations that broke on restart because a key component of the system was broken across PBCs in output coordinates. The user's guide for NAMD could strongly advise against using the wrapAll option.

**Comment by jhenin** [comment]
A complex, but possible way to deal with some of these issues on the colvars side would be to calculate minimum image distances using rotated PBC vectors when the coordinates involved are rotated.

**Comment by giacomofiorin** [comment]
How about we also check for discontinuities in the value of the variable
when reading a state file?

This should catch other errors as well.

The problem is that the user may not read the warning and this should
perhaps be an error?

**Comment by giacomofiorin** [comment]
More on that, how about the following addition to `colvarproxy_namd::setup()`?
```
if (simparams->wrapAll) {
  // Issue a stern warning or raise an error condition
}
```

**Comment by jhenin** [comment]
Yes and yes on those two ideas! Maybe not an error on wrapAll, but definitely a warning.

A suggestion from Grace: "I think CHARMM-GUI outputs .config files with wrapall on - maybe if Wonpil agreed to turn that off it would cut down on the incidence rate of issues with colvars ?"

**Comment by gbrannigan** [comment]
Adding a hint about post-processing wrapping, i.e. qwrap, would probably increase the number of users who complied with the wrapall off recommendation.    (I'm guessing there is some reason colvars can't just automatically qunwrap the system during initialization?)

**Comment by giacomofiorin** [comment]
Well, now, I was not going to mention CHARMM GUI explicitly, but since you did: one of its drawbacks that we have to accept is dealing with users who are less trained (and less interested) in the technical details of MD. 

A warning on wrapAll  may be good to print in the standard output, but few read that.  It may be also good to print a comment line in the Colvars output files (traj and pmf), but that's only marginally better. 

I highly that disabling wrapAll in CHARMM-GUI is an option, given that its main use is precisely to serve people who are not familiar with PBCs (and could be confused by unwrapped coordinates and the famous "protein leaving the box" condition).

So if the problem is specifically that, I see not many options aside from disabling wrapAll every time Colvars is on (see commit e162655).  If you agree, I'll send the patch to Jim for consideration, and to Wonpil for knowledge.

**Comment by jhenin** [comment]
Changing the NAMD setting forcibly seems like a kind of nuclear option... It might lead to panicked reports of "protein leaving the box even though I enabled wrapAll".

In my opinion, your proposed check for jumps upon restart would already go a long way towards catching problems early, and would not change an otherwise expected behavior of NAMD.

**Comment by giacomofiorin** [comment]
I forgot to include the unwrapping issue in my reply.  Unfortunately NAMD does not have that ability, only the wrapping one, so we cannot cleanly trigger an unwrapping from Colvars.

**Comment by giacomofiorin** [comment]
@jhenin yes it is a nuclear option, so I'm submitting it for review before pushing it outside.

My problem is that a warning would be largely ignored, especially if issued in the 20th or 50th run of a long chain.

Do you feel strongly against making the check for jumps an error, and not just a warning?

**Comment by jhenin** [comment]
I would absolutely make a jump an error condition. Now what should the criterion for a jump be? We can't use relative change since colvars can be negative or positive or zero. I think it only needs to account for the finite precision of coordinate files, and whatever uncertainty that may give on colvars. Maybe an absolute change of 1e-2, or 1e-2 * width, if that is available? We can assume that colvars are very roughly on the order of one, but we can't throw errors all the time if someone comes up with a variable that takes large values.

**Comment by giacomofiorin** [comment]
With a smallish threshold like that, it's safe to say that the same magnitude of jump for two variables that have otherwise similar widths, could mean a real error in one and just low precision of the input files in the other.

I guess the error condition should be possible to override somehow?

**Comment by jhenin** [comment]
Then let's have fairly high thresholds, at the risk of missing some errors, but minimizing false positives. If a false positive does occur, and provided it's very rare, the workaround would be editing the state file manually to match the new colvar value of the restart coordinates.

**Comment by giacomofiorin** [comment]
Nuclear option disabled (commit e752f9db67720492a2c766f26a918c48b7e8bcdc), wrapAll warning implemented (commit 70f913c2a3418ecac4a8f8b858ae2a7f056d6cd3) and error upon restart implemented (commit 27a2cca695e921f192fd487286f70fd19fc90726)
